agent/containers/images/image-test-script.sh

```diff
@@ -74,15 +136,20 @@ anothertool=y
 while [ $anothertool == "y" ]
 do
     read -p "Tool name (type 'help' to see options): " tool
-    if [ $tool == "help" ]
+    if [ -z "$tool" ]
     then
-        printf -- "\nAvailable tools:\n"
-        python3 -c "import sys, json; meta = json.load(open(sys.argv[1])); print('  Transient:', *[f'\t{tool}' for tool in meta['transient'].keys()], '  Persistent:', *[f'\t{tool}' for tool in meta['persistent'].keys()], sep='\n')" /opt/pbench-agent/tool-scripts/meta.json
+        printf -- "Tool name empty. Please write a tool name.\n"
     else
-        printf -- "\npbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis\n"
-        pbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis
-        sleep 1
-        read -p "Another tool? (y/n)" anothertool
+        if [ $tool == "help" ]
+        then
+            printf -- "\nAvailable tools:\n"
+            python3 -c "import sys, json; meta = json.load(open(sys.argv[1])); print('  Transient:', *[f'\t{tool}' for tool in meta['transient'].keys()], '  Persistent:', *[f'\t{tool}' for tool in meta['persistent'].keys()], sep='\n')" /opt/pbench-agent/tool-scripts/meta.json
+        else
+            printf -- "\npbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis\n"
+            pbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis
```

Comment:
It would be better to create the command in a variable, print the variable, and then execute it -- that way what the user sees cannot be different from what gets run:
```
            cmd = "pbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis"
            printf -- "\n${cmd}\n"
            $cmd
```